src/components/MessageHistory.tsx

```diff
@@ -0,0 +1,61 @@
+import * as React from "react";
+
+import User from "../util/User";
+
+const messageHistoryStyles = require("./MessageHistory.css");
+
+export interface IMessageHistoryEntry {
+  user: User;
+  message: string;
+  timestamp: Date;
+}
+
+interface IMessageHistoryProps {
+  messages: IMessageHistoryEntry[];
+}
+
+export default class MessageHistory extends React.Component<IMessageHistoryProps, {}> {
+
+  private static dateToTimeAgo(date: Date): string {
```

Comment:
Extract those functions (see also StackedDate) to utils? We might need them later in other views and they are somewhat static.